--- a/gnucash/import-export/qif-imp/qif-to-gnc.scm        2024-12-13 17:46:48.000000000 -0500
+++ b/gnucash/import-export/qif-imp/qif-to-gnc.scm        2025-03-02 09:42:01.075683700 -0500
@@ -434,7 +434,7 @@
             ;; there aren't at least 2 splits and that will cause a
             ;; UAF in xaccTransRecordPrice. See https://bugs.gnucash.org/show_bug.cgi?id=799420
             (let ((splits (qif-xtn:splits xtn)))
-              (if (not (or (qif-xtn:mark xtn) (or (null? splits) (null? (cdr splits)))))
+              (if (not (or (qif-xtn:mark xtn) (null? splits)))
                 ;; Convert into a GnuCash transaction.
                 (let ((gnc-xtn (xaccMallocTransaction
                                 (gnc-get-current-book))))
