--- a/seek_set.c 2001-02-11 16:11:45.000000000 -0500
+++ b/seek_set.c 2025-01-04 20:51:06.232244496 -0500
@@ -1,5 +1,6 @@
 #include <sys/types.h>
 #include "seek.h"
+#include <unistd.h>
 
 #define SET 0 /* sigh */
 
--- a/chkshsgr.c 2025-01-04 20:54:18.521372127 -0500
+++ b/chkshsgr.c 2025-01-04 20:55:10.183449166 -0500
@@ -1,4 +1,6 @@
 #include "exit.h"
+#include <unistd.h>
+#include <grp.h>
 
 int main()
 {

--- a/prot.c     2025-01-04 20:56:44.427115321 -0500
+++ b/prot.c     2025-01-04 20:57:10.399645025 -0500
@@ -1,5 +1,7 @@
 #include "hasshsgr.h"
 #include "prot.h"
+#include <unistd.h>
+#include <grp.h>
 
 int prot_gid(int gid)
 {

