diff --git a/tinydns-conf.c b/tinydns-conf.c
index d3a4ce5..8f43519 100644
--- a/tinydns-conf.c
+++ b/tinydns-conf.c
@@ -46,7 +46,7 @@ int main(int argc,char **argv)

   start("run");
   outs("#!/bin/sh\nexec 2>&1\nexec envuidgid "); outs(user);
-  outs(" envdir ./env softlimit -d300000 ");
+  outs(" envdir ./env softlimit -d4500000 ");
   outs(auto_home); outs("/bin/tinydns\n");
   finish();
   perm(0755);
